packages/tenant-process/test/tenant.integration.test.ts

```diff
@@ -607,6 +607,204 @@ describe("Integration tests", () => {
         );
       });
     });
+    describe("internalAssignCertifiedAttribute", async () => {
+      const correlationId = generateId();
+      const requesterTenant: Tenant = {
+        ...mockTenant,
+        features: [
+          {
+            type: "PersistentCertifier",
+            certifierId: generateId(),
+          },
+        ],
+        updatedAt: currentDate,
+        name: "A requesterTenant",
+        externalId: {
+          origin: generateId(),
+          value: "1234567",
+        },
+      };
+      const attribute: Attribute = {
+        name: "an Attribute",
+        id: unsafeBrandId(requesterTenant.id),
+        kind: "Certified",
+        description: "an attribute",
+        creationTime: new Date(),
+        code: "123456",
+        origin: requesterTenant.externalId.origin,
+      };
+
+      it("Should add the certified attribute if certifiedTenantAttribute doesn't exist", async () => {
+        await addOneAttribute(attribute, attributes);
+        await addOneTenant(requesterTenant, postgresDB, tenants);
+        await tenantService.internalAssignCertifiedAttribute(
+          requesterTenant.externalId.origin,
+          requesterTenant.externalId.value,
+          attribute.origin!,
+          attribute.code!,
+          correlationId
+        );
+        const writtenEvent: StoredEvent | undefined =
+          await readLastEventByStreamId(
+            requesterTenant.id,
+            eventStoreSchema.tenant,
+            postgresDB
+          );
+        if (!writtenEvent) {
+          fail("Update failed: tenant not found in event-store");
+        }
+        expect(writtenEvent).toMatchObject({
+          stream_id: requesterTenant.id,
+          version: "1",
+          type: "TenantCertifiedAttributeAssigned",
+        });
+        const writtenPayload = protobufDecoder(
+          TenantCertifiedAttributeAssignedV2
+        ).parse(writtenEvent?.data);
+
+        const updatedTenant: Tenant = {
+          ...requesterTenant,
+          attributes: [
+            {
+              id: unsafeBrandId(attribute.id),
+              type: "PersistentCertifiedAttribute",
+              assignmentTimestamp: new Date(
+                Number(
+                  (
+                    writtenPayload.tenant!.attributes[0].sealedValue as {
+                      oneofKind: "certifiedAttribute";
+                      certifiedAttribute: TenantCertifiedAttributeV2;
+                    }
+                  ).certifiedAttribute.assignmentTimestamp
+                )
+              ),
+            },
+          ],
+          kind: fromTenantKindV2(writtenPayload.tenant!.kind!),
+          updatedAt: new Date(Number(writtenPayload.tenant?.updatedAt)),
+        };
+        expect(writtenPayload.tenant).toEqual(toTenantV2(updatedTenant));
+      });
+      it("Should add the certified attribute if certifiedTenantAttribute exist", async () => {
```

Comment:
In this test it's not clear that the attribute should have been revoked in order to reassign it.

I suggest to remove `revocationTimestamp` from `getMockCertifiedTenantAttribute` in order to always get a valid `Attribute` and then set the revocation where needed